Dockerfile

```diff
@@ -16,6 +16,7 @@ RUN yarn install
 
 ENV PROVIDER PROVIDER
 ADD contracts contracts
+ADD tests tests
 ADD saddle.config.js saddle.config.js
```

Comment:
Each of these ADDs actually adds a layer (slows down building, pushing), maybe we could copy them all on one line?

Reply:
I can look into that.

Reply:
I replace add to copy statements. 
To copy multiple directories in a single command (not the content), they need to have common parent dir, and copy that parent. 
Also we can replace 3 `COPY `statement here to `COPY . .`, but a lot of additional files should be added to .dockerignore.